File: model2/model2.py
#
# Bakalarska prace
# Robin Vyslouzil
# xvyslo05@stud.fit.vutbr.cz
# Model kruhoveho testu pro nastroj SciPy
#

import numpy
import time
from scipy.integrate import quad
from scipy.integrate import odeint
import matplotlib.pyplot as pyplot

startTime = time.time()

def model2(y, t):
    dxdt = y[1]
    dydt = -y[0]

    result = [dxdt, dydt]

    return result


y0 = [100, 0]

t = numpy.linspace(0, 10000, 5000)

y = odeint(model2, y0, t)

pyplot.plot(t, y)

endTime = time.time() - startTime

print(endTime)


# pyplot.show()

File: model2/model2.m
/*
Bakalarska prace
Robin Vyslouzil
xvyslo05@stud.fit.vutbr.cz
Model kruhoveho testu pro nastroj MATLAB
*/

inittime = clock;

syms x(t) y(t)

ode1 = diff(x) == y;
ode2 = diff(y) == -x;

odes=[ode1; ode2]

S = dsolve(odes)

xSol(t) = S.x
ySol(t) = S.y

cond1 = x(0) == 0;
cond2 = y(0) == 100;
conds = [cond1; cond2];
[xSol(t), ySol(t)] = dsolve(odes,conds)

endtime = clock - inittime;
disp(endtime);
%
% fplot(ySol)
% legend('ySol','Location','best')
